crates/network/src/metric_definitions.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use metrics::{
+    counter, describe_counter, describe_gauge, describe_histogram, gauge, histogram, Counter,
+    Gauge, Histogram, Unit,
+};
+use once_cell::sync::Lazy;
+
+const NETWORK_CONNECTION_CREATED: &str = "restate.network.connection_created.total";
+const NETWORK_CONNECTION_DROPPED: &str = "restate.network.connection_dropped.total";
+const NETWORK_ONGOING_DRAINS: &str = "restate.network.ongoing_drains";
+const NETWORK_MESSAGE_SENT: &str = "restate.network.message_sent.total";
+const NETWORK_MESSAGE_RECEIVED: &str = "restate.network.message_received.total";
+
+const NETWORK_CONNECTION_SEND_DURATION: &str = "restate.network.connection_send_duration.seconds";
+const NETWORK_MESSAGE_PROCESSING_DURATION: &str =
+    "restate.network.message_processing_duration.seconds";
+
+pub static INCOMING_CONNECTION: Lazy<Counter> =
+    Lazy::new(|| counter!(NETWORK_CONNECTION_CREATED, "direction" => "incoming"));
+
+pub static OUTGOING_CONNECTION: Lazy<Counter> =
+    Lazy::new(|| counter!(NETWORK_CONNECTION_CREATED, "direction" => "outgoing"));
+
+pub static CONNECTION_DROPPED: Lazy<Counter> = Lazy::new(|| counter!(NETWORK_CONNECTION_DROPPED));
+pub static ONGOING_DRAIN: Lazy<Gauge> = Lazy::new(|| gauge!(NETWORK_ONGOING_DRAINS));
```

Comment:
Storing the metrics in globals is done to avoid the overhead of repeated registrations of the metrics, right?

Reply:
It doesn't save much but in counters where we don't have dynamic labels, I'd prefer to simplify the call site this way.